File: Gombang/Weekly Contest 403/3195_Find_the_Minimum_Area_to_Cover_All_Ones_I.cs
public class Solution
{
    public int MinimumArea(int[][] grid)
    {

        // 가로로 보았을 때, 1의 인덱스가 가장 작은 값과 가장 큰 값을 담을 변수
        int horizontalFirstOneIndex = int.MaxValue;
        int horizontalLastOneIndex = int.MinValue;

        // 세로로 보았을 때, 1의 인덱스가 가장 작은 값과 가장 큰 값을 담을 변수
        int verticalFirstOneIndex = int.MaxValue;
        int verticalLastOneIndex = int.MinValue;

        for (int i = 0; i < grid.Length; i++)
        {
            for (int j = 0; j < grid[0].Length; j++)
            {
                if (grid[i][j] == 1)
                {
                    if (horizontalFirstOneIndex > j) horizontalFirstOneIndex = j;
                    if (horizontalLastOneIndex < j) horizontalLastOneIndex = j;

                    if (verticalFirstOneIndex > i) verticalFirstOneIndex = i;
                    if (verticalLastOneIndex < i) verticalLastOneIndex = i;
                }
            }
        }

        int width = horizontalLastOneIndex - horizontalFirstOneIndex + 1;
        int height = verticalLastOneIndex - verticalFirstOneIndex + 1;

        return width * height;
    }
}

File: Seori/Weekly Contest 403/3195. Find the Minimum Area to Cover All Ones I.py
class Solution:
    def minimumArea(self, grid: List[List[int]]) -> int:
        width, height = len(grid[0]), len(grid)
        
        # [1] 1이 있는 top, bottom 인덱스를 찾는다. Find the top and bottom indexes where 1 exists.
        top, bottom = 0, 0
        for i in range(height):
            if 1 in grid[i]:
                top = i
                break

        for i in range(height-1, -1, -1):
            if 1 in grid[i]:
                bottom = i
                break
        
        # [2] grid을 회전시켜서 1이 있는 left, right 인덱스를 찾는다. Rotate the grid to find the left and right indexes where 1 exists.
        left, right = 0, 0
        grid_rotate = list(map(list, zip(*reversed(grid))))
        for i in range(width):
            print(i)
            if 1 in grid_rotate[i]:
                left = i
                break

        for i in range(width-1, -1, -1):
            if 1 in grid_rotate[i]:
                right = i
                break
        
        # [3] 넓이를 계산하여 반환한다. Calculate the area and return.
        return (bottom - top + 1) * (right - left + 1)